dataclients/kubernetes/crd.md

```diff
@@ -0,0 +1,898 @@
+# RouteGroup Proposal
+
+Date: 2019-09-04
+Status: open
+Owners: @aryszka @szuecs
+
+This proposal will be kept up to date, if you like to discuss it
+please use https://github.com/zalando/skipper/issues/660
+
+Disclaimer:
+Skipper's `kubernetes` dataclient will also support Ingress in the
+future.
+
+## Purpose
+
+The purpose of the document is to define a RouteGroup CRD, that allows
+configuring Skipper for Kubernetes such that it benefits from all the
+Skipper features in a straightforward way.
+
+## Problem statement
```

Comment:
@szuecs I still think the problem statement is too "weak" for the big change we want to make.
It focuses on a specific use which may be the initial motivation for this change, but it's not the only reason afaik.

IMO the problem is the following (and possible more)

* Multiple ingresses are complicated for users. It's easier for users to create overlapping ingresses when they can't define everything in a single resource.
* Orchestrating traffic switching is complicated when multiple ingresses are involved.
* Configuring everything via annotations is not as nice as configuring via fields in the spec.

Additionally we should make it clear why we don't go with one of the other designs already in the community like SMI. Maybe this doesn't belong in the problem statement but at least in the background information.

Reply:
Let's talk, I think I covered all your points already in the last section of the problem statement.

I started a longer explanation what I think is wrong in x and y, but there was a good point of being nicer and telling what we want to achieve: better skipper integration in Kubernetes.

Reply:
I think I covered now everything more clearly.
I also added in the bottom a  "why not adapt X" section